src/elastio-snap.c

```diff
@@ -5429,6 +5518,8 @@ static asmlinkage long umount_hook(char __user *name, int flags){
 	kfree(buff_dev_name);
 
 	ret = handle_bdev_mount_nowrite(name, flags, &idx);
+	bdev_switch_ownership(name, flags, OWNERSHIP_TO_PARENT);
```

Comment:
Should we care about the return value here?

Reply:
Don't think so. Here we unconditionally change the ownership _before_ umount, and if it wasn't successful, re-gain it back in `post_umount_check()`

Reply:
Ok.